siliconcompiler/package/__init__.py

```
@@ -545,15 +546,84 @@ def resolve(self) -> Union[str, Path]:
             except BaseException:
                 # Exception occurred, so need to cleanup
                 try:
+                    # Make writable first, in case cache was partially made read-only
+                    try:
+                        self._make_writable(self.cache_path)
+                    except OSError as e:
+                        self.logger.warning(f"Could not make cache writable before cleanup: {e}")
                     shutil.rmtree(self.cache_path)
                 except BaseException as cleane:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `BaseException`

(BLE001)